src/inmanta/protocol/openapi/converter.py

```diff
@@ -411,10 +412,49 @@ def _build_responses(self, url_method_properties: MethodProperties) -> Dict[str,
 
     def _build_return_value_wrapper(self, url_method_properties: MethodProperties) -> Optional[Dict[str, MediaType]]:
         return_type = inspect.signature(url_method_properties.function).return_annotation
-        if return_type is not None and return_type != inspect.Signature.empty:
-            return_properties: Optional[Dict[str, SchemaBase]] = None
+
+        if return_type is None or return_type == inspect.Signature.empty:
+            return None
+
+        return_properties: Optional[Dict[str, SchemaBase]] = None
+
+        if isinstance(return_type, GenericMeta) and return_type.__origin__ == ReturnValue:
```

Comment:
Is this the correct way of unwrapping? Should the introspection library be used?

Reply:
There is indeed support for this in a library. Apart just the fact that this abstraction exists, another great reason to use it is that I just found out that `GenericMeta` does not exist in Python 3.9 (only tested 3.6 and 3.9).
- `typing_inspect.is_generic_type(t)` returns `True` for any generic type, with or without args (`List` or `List[int]`)
- `typing_inspect.get_origin(t)` returns the origin type. In python 3.9 this seems to return `None` for generic types without parameters, so it might be good to  check for this condition.
- `typing_inspect.get_args(t)` returns a tuple of type arguments.

Reply:
I have been looking for those in the inspect package, didn't consider there might be one for typing.  Thanks!